zerver/views/users.py

```diff
@@ -114,6 +115,23 @@ def update_user_backend(request, user_profile, email,
 
     return json_success()
 
+@has_request_variables
+def update_buddy_list(request, user_profile, user_id=REQ(validator=check_string),
+                      buddy_id=REQ(validator=check_string), should_add=REQ(validator=check_string)):
+    # type: (HttpRequest, UserProfile, str, str, str) -> HttpResponse
+    try:
```

Comment:
We need to check that user_profile.id == user_id for the incoming requesting parameters, unless user_profile is an admin.  Since there's not a huge need for admins to be able to modify buddy lists, it might make sense to omit the `user_id` parameter here and just assume we're always adding a buddy for our current user.